enaml/src/colorext.cpp

```diff
@@ -160,9 +160,30 @@ Color_getset[] = {
 };
 
 
+static PyObject*
+Color__reduce__( Color* self, void* context )
+{
+    PyObject* cls = PyObject_Type(reinterpret_cast<PyObject*>( self ));
```

Comment:
`PyObject_Type` returns a new reference to the type object. The problem is that `Py_BuildValue` also increments the reference of the object passed with the `O` flag. So, this method is leaking a reference to the type object.

The proper way to do this (which only borrows a reference) would be:
`PyObject* cls = reinterpret_cast<PyObject*>( self->ob_type );`

Reply:
Nice catch !

Reply:
I had tried something like this originally but it is giving an error
```
enaml/src/colorext.cpp: In function ‘PyObject* Color__reduce__(Color*, void*)’:
enaml/src/colorext.cpp:166:56: error: ‘struct Color’ has no member named ‘ob_type’
     PyObject* cls = reinterpret_cast<PyObject*>( self->ob_type );
                                                        ^
error: command 'gcc' failed with exit status 1
```

This works

```cpp

static PyObject*
Color__reduce__( Color* self, void* context )
{
    PyObject* obj = reinterpret_cast<PyObject*>( self );
    if ( !obj )
        return 0;
    uint32_t a = ( self->argb >> 24 ) & 0xFF;
    uint32_t r = ( self->argb >> 16 ) & 0xFF;
    uint32_t g = ( self->argb >> 8 ) & 0xFF;
    uint32_t b = self->argb & 0xFF;
    return Py_BuildValue( "O (iiii)", obj->ob_type, r, g, b, a);
}



```